api/src/main/java/org/apache/cloudstack/diagnostics/RetrieveDiagnosticsService.java

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.diagnostics;
+
+import com.cloud.exception.InvalidParameterValueException;
+import com.cloud.utils.Pair;
+import com.cloud.utils.component.Manager;
+import com.cloud.utils.component.PluggableService;
+import org.apache.cloudstack.api.command.admin.diagnostics.RetrieveDiagnosticsCmd;
+import org.apache.cloudstack.api.response.RetrieveDiagnosticsResponse;
+import org.apache.cloudstack.config.Configuration;
+import org.apache.cloudstack.framework.config.ConfigKey;
+
+import javax.naming.ConfigurationException;
+import java.util.List;
+import java.util.Map;
+
+
+public interface RetrieveDiagnosticsService extends Manager, PluggableService {
+
+    RetrieveDiagnosticsResponse getDiagnosticsFiles(final RetrieveDiagnosticsCmd cmd) throws InvalidParameterValueException, ConfigurationException;
+
+    ConfigKey<?>[] getConfigKeys();
```

Comment:
this method shouldn't be declared here. It is part of interface Configurable.

Reply:
Removed ConfigKey<?> getConfigKeys() from ServiceImpl

Reply:
It should remain in the ServiceImpl, @charles-phiri . But it shouldn't be in this interface. It is already defined in Configerable.